src/lib/support/PersistedCounter.h

```diff
@@ -197,7 +197,7 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
             ReturnErrorOnFailure(err);
         }
 
-        if (size != sizeof(valueLE))
+        if (size > sizeof(valueLE))
```

Comment:
Why is this OK?  If `size < sizeof(valueLE)`, then some of the bytes of the value did not get initialized by the read from storage.  Why is it OK to use the value?

Also, since we stored `sizeof(valueLE)` bytes to start with, why would storage be returning some smaller number of bytes here?  That's a violation of the storage API contract!

Reply:
The variable `valueLE` is initialized to zero, so if the upper bits are left untouched, the returned value should still be correct. Unless, for some reason, `SyncGetKeyValue` is adding random bits beyond the size indicated by the returned `size` variable.

Reply:
Though it makes sense that `SyncGetKeyValue` should return a `size` equal to `sizeof(valueLE)`, I don't see anything in this context that ensures such behavior.

Reply:
Agreed. This change shall be reverted.

Reply:
#18115 reverts the change and changes the key as to avoid any conflicts